packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -67,24 +66,28 @@ class TransactionService {
     this.currentTransaction = value
   }
 
-  createTransaction(name, type, options) {
-    var tr = new Transaction(name, type, options)
-    this.setCurrentTransaction(tr)
-    if (options.checkBrowserResponsiveness) {
-      this.startCounter(tr)
+  ensureRespInterval() {
+    let currentTr = this.getCurrentTransaction()
+
+    const clearRespInterval = () => {
+      window.clearInterval(this.respIntervalId)
+      this.respIntervalId = undefined
     }
-    return tr
-  }
 
-  startCounter(transaction) {
-    transaction.browserResponsivenessCounter = 0
-    const id = setInterval(function() {
-      if (transaction.ended) {
-        window.clearInterval(id)
-      } else {
-        transaction.browserResponsivenessCounter++
+    if (currentTr && currentTr.options.checkBrowserResponsiveness) {
+      if (typeof this.respIntervalId === 'undefined') {
+        this.respIntervalId = setInterval(() => {
+          let tr = this.getCurrentTransaction()
+          if (tr) {
+            tr.browserResponsivenessCounter++
+          } else {
+            clearRespInterval()
+          }
+        }, BROWSER_RESPONSIVENESS_INTERVAL)
       }
-    }, BROWSER_RESPONSIVENESS_INTERVAL)
+    } else if (typeof this.respIntervalId !== 'undefined') {
```

Comment:
This else if block seems like a dead code to me. Am i missing something?

Reply:
Since the `checkBrowserResponsiveness` can change (e.g. via redefined) we might end up in this code path.